generators/new_collateral/types/collateral.go

```diff
@@ -17,6 +17,24 @@ func NewCollateral(name, version string) Collateral {
 	}
 }
 
+func (c Collateral) formattedLowerCaseName() string {
+	return strings.ToLower(strings.Replace(c.Name, "-", "_", -1))
+}
+
+func (c Collateral) formattedUpperCaseName() string {
+	return strings.ToUpper(strings.Replace(c.Name, "-", "_", -1))
+}
+
+func (c Collateral) FormattedLowerCaseNameWithoutVersionLetter() string {
```

Comment:
These two methods probably aren't super future proof since they are expecting that all collateral names be "NAME" or "NAME_LETTER" and just chop the letter piece off, and this probably isn't guaranteed. But I think using this method will it easier to add new collaterals to the system in the short term, and then maybe we can re-think this change as part of https://makerdao.atlassian.net/browse/VDB-1704

Reply:
Makes sense to me. Maybe worth updating that story? Thinking these changes prevent us from running into cases where it generates `median_eth_b`, but the new risk would be chopping a multi-word collateral name - is that right? Like, `collateral_number_one_a` would end up parsing to `median_collateral` instead of `median_collateral_number_one` - if I follow.

Reply:
Yep, you got it - this wouldn't work if the collateral has a multi-word name. Updated VDB-1704 to mention this ✔️